main.c

```diff
@@ -507,15 +507,39 @@ static void HandleFalling()
 
 int main()
 {
-    int keysPressed[6], close = 0;
+    int keysPressed[6] = {0,0,0,0,0,0}; //Init key states to not pressed.
+    int close = 0;
     LoadMapFile();
     SDL_Init(SDL_INIT_EVERYTHING);
+    
     window = SDL_CreateWindow("2.5d Engine David", 0, 0, WIDTH, HEIGHT, SDL_WINDOW_MAXIMIZED);
-    surface = SDL_GetWindowSurface(window);
+    
+    if (window == NULL)
+    {
+        printf("SDL_CreateWindow failed: %s\n", SDL_GetError());
+        exit(1);
+    }
+    
+    //surface = SDL_GetWindowSurface(window); Should apparently be used with SDL_CreateRenderer!
```

Comment:
This comment should read: Should apparently NOT be used with SDL_CreateRenderer!